src/components/BookList.js

```diff
@@ -1,26 +1,25 @@
 import React from 'react';
-import { removeItem } from '../redux/books/bookSlice'
+import { deleteBook, getBooks} from '../redux/books/bookSlice'
 import { useDispatch } from 'react-redux';
-// import {store} from '../redux/store'
 
-const BookList = ({ editBook, id, title, author , category}) => {
-const dispatch = useDispatch();
-  const handleRemoveItem = (id) => {
-    dispatch(removeItem(id));
+const BookList = ({ book, id }) => {
+  const dispatch = useDispatch();
+  const handleRemoveItem = async (item, id) => {
```

Comment:
- [ ] You should use `useEffect` when dispatching your `getBooks()` function here. It should look something like this:
```jsx
  useEffect(() => {
    dispatch(fetchBooks())
  }, [])
```